packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -291,14 +290,20 @@ export class DataStores implements IDisposable {
         return this.aliasMap.get(id) !== undefined || this.contexts.get(id) !== undefined;
     }
 
-    public bindFluidDataStore(fluidDataStoreRuntime: IFluidDataStoreChannel): void {
-        const id = fluidDataStoreRuntime.id;
+    /**
+     * Makes the data store with the given id visible in the container graph by moving the data store context from
+     * unbound to bound list. This data store can now be reached from the root.
+     * @param id - The id of the data store context to make visible.
+     */
+    public makeDataStoreVisible(id: string): void {
```

Comment:
I think this API should be private, as there is no way to directly call it (i.e. container runtime does not call it, and that's the only class that has direct access to this class).

It's minor, but I think we should reduce usage of IDs across our code base, and use objects pretty much everywhere. Maybe other way to say it - other than serialization / deserialization (and thus usage of handles), the system can and should be similar to "raw" JS environment, where objects creates by JS engine do not have id / other kind of identification, it's just an object.
Main reason why I think it's good idea - if we do get to implementing ID compression schema, we would have 2 internal IDs (in addition to aliases). And while I hear there is strong type safety built into existing ID compression logic, it would still be a pain to reason about what ID is used where. Less IDs - better.

Reply:
Yes, you are right. This should be private.